src/features/currentTodo.ts

```diff
@@ -1,32 +1,72 @@
 import { Todo } from '../types/Todo';
 
-// we use string literal as a type to avoid mistype in future
 type RemoveTodoAction = { type: 'currentTodo/REMOVE' };
 
-// payload is a typical name for an action data
 type SetTodoAction = {
   type: 'currentTodo/SET';
   payload: Todo;
 };
 
-// Action creator return type protect us from a mistype
+type SetLoadigAction = {
+  type: 'currentTodo/SET_LOADING';
+  payload: boolean;
+};
+
+type SetErrorAction = {
+  type: 'currentTodo/SET_ERROR';
+  payload: string;
+};
+
 const removeTodo = (): RemoveTodoAction => ({ type: 'currentTodo/REMOVE' });
 
 const setTodo = (todo: Todo): SetTodoAction => ({
   type: 'currentTodo/SET',
   payload: todo,
 });
 
-// These actions will be used in the application
-export const actions = { setTodo, removeTodo };
+const setLoadig = (loading: boolean): SetLoadigAction => ({
+  type: 'currentTodo/SET_LOADING',
+  payload: loading,
+});
 
-type State = Todo | null;
-type Action = SetTodoAction | RemoveTodoAction;
+const setError = (error: string): SetErrorAction => ({
+  type: 'currentTodo/SET_ERROR',
+  payload: error,
+});
 
-const currentTodoReducer = (state: State = null, action: Action): State => {
-  switch (action.type) {
-    // Implement all actions here
+export const actions = { setTodo, removeTodo, setLoadig, setError };
 
+type Action =
+  | SetTodoAction
+  | RemoveTodoAction
+  | SetErrorAction
+  | SetLoadigAction;
+
+type State = {
+  currentTodo: Todo | null;
+  loading: boolean;
+  error: string;
+};
+
+const initialState: State = {
+  currentTodo: null,
+  loading: false,
+  error: '',
+};
+
+const currentTodoReducer = (
+  state: State = initialState,
```

Comment:
That's the case when we should ignore lint error. In such cases it's better to disable the rule for this file:
![image](https://github.com/mate-academy/react_redux-list-of-todos/assets/86923089/c073ba2a-cb3c-48d0-8b7e-f3684631dd46)

FYI